com.microsoft.mrtk.input/Utilities/EyeCalibrationChecker.cs

```diff
@@ -0,0 +1,161 @@
+        /// </summary>
+        [field: SerializeField, Tooltip("For testing purposes, you can manually assign whether eyes are calibrated or not in editor.")]
+        public EyeCalibrationStatus EditorTestIsCalibrated = EyeCalibrationStatus.IsCalibrated;
+
+        #endregion Serialized Fields
+
+        #region Private Fields
+        private EyeCalibrationStatus calibrationStatus;
+
+        /// <summary>
+        /// Tracks whether eyes are present and calibrated. 
+        /// </summary>
+        public EyeCalibrationStatus CalibratedStatus
+        {
+            get => calibrationStatus;
+        }
+
+        private EyeCalibrationStatus prevCalibrationStatus;
+        #endregion Private Fields
+
+        #region Events
+        /// <summary>
+        /// Event fired when IsCalibrated changes from false to true.
+        /// </summary>
+        public UnityEvent OnEyeCalibrationDetected = new UnityEvent();
+
+        /// <summary>
+        /// Event fired when IsCalibrated changes from true to false.
+        /// </summary>
+        public UnityEvent OnNoEyeCalibrationDetected = new UnityEvent();
+        #endregion Events
+
+        #region MonoBehaviour Functions
+        private void Update()
+        {
+            if (Application.isEditor)
+            {
+                calibrationStatus = EditorTestIsCalibrated;
+            }
+            else
+            {
+                calibrationStatus = CheckCalibrationStatus();
+            }
+
+            if (prevCalibrationStatus != calibrationStatus)
+            {
+                if (calibrationStatus == EyeCalibrationStatus.IsCalibrated)
+                {
+                    OnEyeCalibrationDetected.Invoke();
+                }
+                else
+                {
+                    OnNoEyeCalibrationDetected.Invoke();
+                }
+                prevCalibrationStatus = calibrationStatus;
+            }
+        }
+        #endregion MonoBehaviour Functions
+
+        #region Private Functions
+
+        private EyeCalibrationStatus CheckCalibrationStatus()
+        {
+#if WINDOWS_UWP
+            if (MixedReality.OpenXR.PerceptionInterop.GetSceneCoordinateSystem(Pose.identity) is SpatialCoordinateSystem worldOrigin)
+            {
+                SpatialPointerPose pointerPose = SpatialPointerPose.TryGetAtTimestamp(worldOrigin, PerceptionTimestampHelper.FromHistoricalTargetTime(DateTimeOffset.Now));
+                if (pointerPose != null)
+                {
+                    EyesPose eyes = pointerPose.Eyes;
+                    if (eyes != null)
+                    {
+                        // If it's been longer than a second since the last perception snapshot, assume the information has expired.
+                        if ((DateTimeOffset.Now - eyes.UpdateTimestamp.TargetTime.DateTime).TotalSeconds > 1)
+                        {
+                            return EyeCalibrationStatus.IsExpired;
```

Comment:
don't think IsExpired provides much value. In this case eyes should be considered NotCalibrated